More documentation touchups

gordonhart opened the pull request:
Fixes KOL-2499
Fixes KOL-2511
Fixes KOL-2521
Fixes KOL-2524
Fixes KOL-2525
Fixes KOL-2526
Fixes KOL-2513
Fixes KOL-2523

gordonhart committed:
Fix numbering on workflow/index.md

gordonhart committed:
Remove buggy navigation.instant feature that breaks forward/back in some circumstances

gordonhart committed:
Touch up 'Building a Workflow' tutorial

gordonhart committed:
Override card grid interactivity styles (cards are not interactive)

gordonhart committed:
Use HTTPS to clone in quickstart; add image to note on 'Building a Workflow' guide

munkyshi commented on docs/building-a-workflow.md:
```diff
@@ -214,13 +224,13 @@ With basic tests defined for the 300-W dataset, we're almost ready to start test
-performance under a variety of conditions.
-For this keypoints example, perhaps we want to compute performance at a few different NME threshold values, as this
-threshold drives the `alignment_failure` metric.
+Evaluators can have arbitrary configurationo [(`EvaluatorConfiguration`)][kolena.workflow.EvaluatorConfiguration],
```
sp: configuration

munkyshi commented:
Feels bad to say, but it now feels like the FAQs are a bit too drab 😅 
<img width="1333" alt="Screenshot 2023-06-14 at 23 25 11" src="https://github.com/kolenaIO/kolena/assets/2481598/b14aa5ba-394d-40cb-bed0-c3359047c2a4">

munkyshi commented:
Unrelated to this PR specifically, but the help icon makes me feel like it's the unrecognized unicode character � (especially when repeated the number of times it is on the FAQ page)

gordonhart committed:
Add custom 'FAQ' admonition type

gordonhart commented:
Related to your comments on FAQs @munkyshi — I've added a custom FAQ admonition type with styles matching table styles. Also repurposed the flag icon for FAQs, and changed the "core concepts" icon to something thematically similar to the flag.

Let me know what you think.

munkyshi reviewed (approved):
Looks good!
